src/analysis/analyzer.ts

```diff
@@ -61,6 +64,10 @@ export class Analyzer extends AnalyzerGen implements vs.Disposable {
 		this.serverSetSubscriptions({
 			subscriptions: ["STATUS"]
 		});
+
+		// Hook error subscriptions so we can try and get diagnostic info if this happens.
+		this.registerForServerError(e => this.requestDiagnosticsUpdate());
```

Comment:
Just wanted to say, this is a pretty cool way to collect diagnostics. I'd taken a stab at a PR like this when I filed the related issue, but got hung up on trying to get a `timeout()` method working in typescript, and eventually abandoned the PR.